web/src/components/core/NumericTextInput.jsx

```diff
@@ -0,0 +1,63 @@
+
+/*
+ * Copyright (c) [2023] SUSE LLC
+ *
+ * All Rights Reserved.
+ *
+ * This program is free software; you can redistribute it and/or modify it
+ * under the terms of version 2 of the GNU General Public License as published
+ * by the Free Software Foundation.
+ *
+ * This program is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License for
+ * more details.
+ *
+ * You should have received a copy of the GNU General Public License along
+ * with this program; if not, contact SUSE LLC.
+ *
+ * To contact SUSE LLC about this file by physical or electronic mail, you may
+ * find current contact information at www.suse.com.
+ */
+
+import React from "react";
+import { TextInput } from "@patternfly/react-core";
+import { noop } from "~/utils";
+
+/**
+ * Callback function for notifying a valid input change
+ *
+ * @callback onChangeFn
+ * @param {string|number} the input value
+ * @return {void}
+ */
+
+/**
+ * Helper component for having an input text limited to not signed numbers
+ * @component
+ *
+ * Based on {@link PF/TextInput https://www.patternfly.org/v4/components/text-input}
+ *
+ * @note It allows empty value too.
+ *
+ * @param {object} props
+ * @param {string|number} props.value - the input value
+ * @param {onChangeFn} props.onChange - the callback to be called when the entered value match the input pattern
+ * @param {object} props.textInputProps - @see {@link https://www.patternfly.org/v4/components/text-input/#textinput}
+ *
+ * @returns {ReactComponent}
+ */
+export default function NumericTextInput({ value = "", onChange = noop, ...textInputProps }) {
```

Comment:
We could go further and do this helper more generic by receiving the pattern via prop. But let's wait until we really need it. Do you agree?

BTW, we are not using directly the HTML [pattern](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/pattern) attribute because it's not intended for limiting the input from the user but for performing validations before send the form. But since we're preventing the form submission, it is not as direct as the proposed solution. To learn more, read https://developer.mozilla.org/en-US/docs/Web/HTML/Constraint_validation#constraint_validation_process

Of course, we can evaluate the use of `pattern` attribute in the future.